src/commands/debug.ts

```diff
@@ -121,6 +121,11 @@ export class DebugCommands {
 				analytics.logDebuggerOpenObservatory();
 			}
 		}));
+		context.subscriptions.push(vs.commands.registerCommand("dart.getObservatoryAddress", () => {
```

Comment:
Sorry about that, I missed package.json when adding files to the patch, which has a template configuration that calls it. I've uploaded it, but I'll take a look at doing it the way you're suggesting here though since that sounds like a simpler flow.

Reply:
> Maybe in resolveDebugConfiguration if we're an attach request and it's not been set we should invoke this automatically

It looks like resolveDebugConfiguration expects to return synchronously, so it wasn't clear to me how to make that work. I did get it working inlined into attachRequest itself though.

I'm not sure if I was missing something, or if launching a dialog box from attachRequest was a terrible idea, so I've kept this part of the patch unchanged for now. Let me know how you'd like me to handle it.

Reply:
You're right that currently our implementation of `resolveDebugConfiguration` is synchronous, but it returns `ProviderResult<T>` which is `T | undefined | null | Thenable<T | undefined | null>` so it can be async (we actually `await` it in the tests, even though don't need to).

I would probably change it to return `Promise<DebugConfiguration>` and mark it as async so we can `await` inside it.

> I'm not sure if I was missing something, or if launching a dialog box from attachRequest was a terrible idea

I don't think so.. As far as I know, MS do similar attaching to node (eg. [see here](https://github.com/Microsoft/vscode-node-debug/blob/master/src/node/extension/configurationProvider.ts#L108-L113)). Based on the comments I think they're showing the prompt before this code (by running a command to pick the process) but there's definitely async code here. I don't think there's anything special about this.

Reply:
Oh, I missed the specifics of the return type; that makes it easy. I've moved it into `resolveDebugConfiguration` now per your suggestion. I also added the port->URL conversion step there using localhost at the default (I don't think there's much value in supporting in launch.json since it's easy to put the full URL if you're hard-coding it once, and it makes the value always match the name.)